### DISCUSSION TITLE
Support `:where(&)` in style selectors

### DISCUSSION THREAD

**Discussion by AndrewLeedham** [vanilla-extract-css/vanilla-extract#928] [Ideas]
https://github.com/vanilla-extract-css/vanilla-extract/pull/204 added support for `&:where(...)` and `&:is(...)`, but I think it would also be useful to allow `:where(&)` to reduce specificity.

You can of course do this with `globalStyle` today, but my use-case includes exporting this from a helper function which returns `rules` to be spread onto an existing `style` tag, using `globalStyle` from within that helper would be a side-effect and would also require passing a a reference to the selector that hasn't been created yet, for example:

```typescript
// myHelper.ts
export default function myHelper(className: string) {
 globalStyle(`:where(${className})`, {
   // Some lower specificity rules.
 });

  return {
    // ...some rules.
  }
}

// example.css.ts
import myHelper from './myHelper';

const { rules } = myHelper(/** rule would need to be passed here, but it hasn't been created yet **/);

const rule = style({
  ...rules
})
```

**Comment by zr00da** [comment]
This syntax should definitely be supported, but I'm wondering if you're maybe after some sort of "default styling that can be easily overridden without overloading selector specificity" with that generic helper? If that's the case and you're using this helper across the codebase a lot you might want to switch to [`@layers`](https://vanilla-extract.style/documentation/api/layer/).
